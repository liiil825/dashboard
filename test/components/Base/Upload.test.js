import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Upload from 'components/Base/Upload';

const findByRef = wrapper => ref => wrapper.findWhere(node => node.getDOMNode() === wrapper.instance()[ref]);

const fileName = 'test.zip';

describe('Base/Upload', () => {
  it('basic render', () => {
    const wrapper = render(<Upload />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('call onChange', () => {
    const wrapper = mount(<Upload />);

    wrapper.findByRef = findByRef(wrapper);

    const input = wrapper.findByRef('fileInput');
    input.simulate('change', { target: { files: [fileName] } });
  });

  it('disabled', () => {
    const wrapper = mount(<Upload disabled />);
    expect(wrapper.prop('disabled')).toBeTruthy();
    wrapper.unmount();
  });
});
