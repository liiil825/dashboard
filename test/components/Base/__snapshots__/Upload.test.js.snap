// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base/Upload basic render 1`] = `
<span
  class="upload"
  role="button"
>
  <input
    style="display:none"
    type="file"
  />
</span>
`;
